### PR TITLE
YAML Editor Snippets

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

Adds snippet support to the YAML editor, along with a few commonly used k8s snippets.

### Screenshots

**Metadata and containers snippets:**
yaml-snippets

Addresses https://jira.coreos.com/browse/CONSOLE-595

### CODE REVIEW

File: frontend/public/module/k8s/autocomplete.ts [hunk 1]
```diff
@@ -9,6 +9,72 @@ import { K8sKind, K8sResourceKind } from '../../module/k8s';
...
+                serviceName: \${2}
+                servicePort: 80
+  `)
+  .set('ports', `
+    ports:
+    - protocol: TCP
+      port: 80
+      targetPort: \${1}
+  `)
+  .map((content, name) => ({name, content: content.split('\n').slice(1).map(s => s.substring(4)).join('\n')} as AceSnippet));
```
Comment (jhadvig) [inline frontend/public/module/k8s/autocomplete.ts]: Really like this these snippets :+1: 
Should this PR just contain basic set of snippets, and we add additional snippets on demand, eg:
``` yaml
parameters:
- name: {
  value: 
  description: 
```

or 

```yaml
triggers:
- imageChange:
    from:
      kind: ImageStreamTag
      name: hello-openshift-binary:latest
  type: ImageChange
```

or those are all we want to have ?

Comment (alecmerdler) [reply frontend/public/module/k8s/autocomplete.ts]: @jhadvig Added those snippets for you. We can make follow-up PRs to add additional snippets as they are needed.

File: frontend/public/module/k8s/autocomplete.ts [hunk 2]
```diff
@@ -9,6 +9,86 @@ import { K8sKind, K8sResourceKind } from '../../module/k8s';
...
+export const snippets = ImmutableMap<string, string>()
+  .set('metadata', `
+    metadata:
+      name: \${1}
+      namespace: \${2}
+      labels: 
+        \${3}
+      annotations:
+        \${4}
+  `)
```
Comment (jhadvig) [inline frontend/public/module/k8s/autocomplete.ts]: So this extra new-line character will be also present in the ace editor:
alec

Not sure if we want those extra new-lines. This applies to all the added snippets.

Comment (alecmerdler) [reply frontend/public/module/k8s/autocomplete.ts]: Fixed it.

Comment (jhadvig) [reply frontend/public/module/k8s/autocomplete.ts]: But the extra line is still there. Shouldn't it be:
```js
  .set('metadata', `
    metadata:
      name: \${1}
      namespace: \${2}
      labels: 
        \${3}
      annotations:
        \${4}`
  )
```

Comment (alecmerdler) [reply frontend/public/module/k8s/autocomplete.ts]: @jhadvig I prefer the backticks be on separate lines to make the template as clean as possible with correct indentation. [This line](https://github.com/openshift/console/pull/202/files#diff-1ff4faa1babdf14e37a86d526de07808R90) performs indentation/newline fixes.

Comment (jhadvig) [reply frontend/public/module/k8s/autocomplete.ts]: oh, I see :+1:

**Pull request merged**